__fixtures__/test-project/web/src/components/Contact/ContactForm/ContactForm.tsx

```diff
@@ -7,10 +7,21 @@ import {
   Submit,
 } from '@redwoodjs/forms'
 
+import type { Contact, UpdateContactInput } from 'types/graphql'
+import type { RwGqlError } from '@redwoodjs/forms'
 
 
-const ContactForm = (props) => {
-  const onSubmit = (data) => {
+
+
+interface ContactFormProps {
+  contact?: Contact
+  onSave: (data: UpdateContactInput, id?: Contact['id']) => void
+  error: RwGqlError
+  loading: boolean
+}
+
+const ContactForm = (props: ContactFormProps) => {
+  const onSubmit = (data: Record<string, any>) => {
```

Comment:
I think we should have much better types here. And to that end I wrote this PR: https://github.com/redwoodjs/redwood/pull/5867

Reply:
Good stuff! I don’t know forms that well, so didn’t dig in.